test/e2e/nodepool_upgrade_test.go

```diff
@@ -121,11 +121,6 @@ func TestReplaceUpgradeNodePool(t *testing.T) {
 		t.Log("NodePool have UpdatingVersionCondition condition")
 	}
 
-	// Wait for at least 1 Node to be unready, so we know the process is started.
-	e2eutil.WaitForNUnReadyNodes(t, ctx, guestClient, 1)
-	t.Log("Upgrade has stated as at least 1 Node to is unready")
-	e2eutil.WaitForNReadyNodes(t, ctx, guestClient, numNodes, hostedCluster.Spec.Platform.Type)
-
 	// Wait for NodePools to roll out the latest version
 	// TODO: Consider doing this in parallel
 	for _, nodePool := range nodePools.Items {
```

Comment:
can we add a wait check after WaitForNodePoolConditionsNotToBePresent to check nodepool.replicas= nodepool.status.replicas? That we way we validate all wanted Nodes remain ready after upgrade.